batcher/aligned-sdk/src/core/types.rs

```diff
@@ -396,6 +400,86 @@ pub enum GetNonceResponseMessage {
     InvalidRequest(String),
 }
 
+pub struct BumpFeeMessage {
+    pub conent: BumpFeeMessageContent,
+    pub signature: Signature,
+}
+
+pub struct BumpFeeMessageContent {
+    pub bump_unit: BumpUnit,
+    pub value: U256,
+    pub proof_qty: usize,
+    pub timestamp: u64,
+}
+
+impl EIP712 for BumpFeeMessageContent {
+    type Error = Eip712Error;
+    fn domain(&self) -> Result<EIP712Domain, Self::Error> {
+        Ok(EIP712Domain {
+            name: Some("Aligned".into()),
+            version: Some("1".into()),
+            chain_id: Some(self.chain_id),
+            verifying_contract: Some("0x00"),
+            salt: None,
+        })
+    }
+    fn type_hash() -> Result<[u8; 32], Self::Error> {
+        let mut hasher = Keccak256::new();
+        hasher.update(NONCED_VERIFICATION_DATA_TYPE);
+        Ok(hasher.finalize().into())
+    }
+
+    pub fn new(bump_unit: BumpUnit, value: U256, proof_qty: usize) -> Self {
+        let now = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .expect("Time went backwards");
```

Comment:
This is actually possible (for how computers process time, not in the physical reality), so I don't recommend crashing. Returning an error if it would be incorrect wouldn't be bad tho.